pkg/server/ptypes/project_template_test.go

```diff
@@ -0,0 +1,153 @@
+package ptypes
+
+import (
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"testing"
+
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+func TestValidateCreateProjectTemplateRequest(t *testing.T) {
+	tests := []struct {
+		name    string
+		req     *pb.CreateProjectTemplateRequest
+		wantErr bool
+	}{
+		{
+			name: "minimum valid request",
+			req: &pb.CreateProjectTemplateRequest{
+				ProjectTemplate: &pb.ProjectTemplate{
+					Name: "test",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "enforces name",
+			req: &pb.CreateProjectTemplateRequest{
+				ProjectTemplate: &pb.ProjectTemplate{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Inherits base validator rules (example: name length)",
+			req: &pb.CreateProjectTemplateRequest{
+				ProjectTemplate: &pb.ProjectTemplate{
+					Name: string(make([]byte, PROJECT_TEMPLATE_NAME_LENGTH+1)),
+				},
+			},
+			wantErr: true,
+		},
```

Comment:
```suggestion
		},
		{
			name: "enforces name to not be empty",
			req: &pb.CreateProjectTemplateRequest{
				ProjectTemplate: &pb.ProjectTemplate{
					Name: " "},
			},
			wantErr: true,
		},

```

Maybe names shouldn't be non-zero length empty strings? I mention this in another comment, it seems like this should error but I don't know if we guard against this anywhere else ¯\_(ツ)_/¯